src/OneExplorer/ConfigObject.ts

```diff
@@ -0,0 +1,194 @@
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as ini from 'ini';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+/**
+ * @brief Parsed .cfg file into an 'obj' which contains fields such as
+ *        'baseModels' or 'derivedModels'
+ *
+ * @usage Direct Parse
+ * ```
+ * const {baseModels, derivedModels} = configObj.parse(uri);
+ * ```
+ *
+ * @usage Create Parsed Config Object
+ *        (use a factory function `createConfigObj`)
+ * ```
+ * const configObj = createConfigObj(uri);
+ * ```
+ */
+export class ConfigObj {
+  /**
+   * source uri of config file
+   */
+  uri: vscode.Uri;
+
+  /**
+   * a raw ini object read from config file
+   */
+  rawObj: object;
+
+  /**
+   * a parsed config object
+   */
+  obj: {
+    baseModels: vscode.Uri[],
+    derivedModels: vscode.Uri[],
+    // TODO: Add more fields
+  };
+
+  private constructor(uri: vscode.Uri, rawObj: object) {
+    this.uri = uri;
+    this.rawObj = rawObj;
+    this.obj = {
+      baseModels: ConfigObj.parseBaseModels(uri, rawObj),
+      derivedModels: ConfigObj.parseDerivedModels(uri, rawObj)
+    };
+  }
+
+  /**
+   * A simple parse function with directly `ConfigObj.obj` fields after parsing
+   * without keeping an ConfigObj instance.
+   *
+   * @param uri
+   * @returns `{baseModels, derivedModels}` or
+   *          `null` if failed to open/parse into ini object
+  }
+   */
+  public static parse(uri: vscode.Uri): {baseModels: vscode.Uri[],
+                                         derivedModels: vscode.Uri[]}|null {
+    const cfgObj = this.createConfigObj(uri);
+
+    if (!cfgObj) {
+      return null;
+    }
+
+    return cfgObj.obj;
+  }
+
+  /**
+   * A factory function to create ConfigObj for the given uri.
+   *
+   * @param uri
+   * @returns `ConfObj` or
+   *          `null`    if failed to open/parse into ini object
+   */
+  public static createConfigObj(uri: vscode.Uri) {
+    const obj = this.importIni(uri.fsPath);
+    if (!obj) {
+      console.error(`Cannot open ${uri.fsPath}`);
+
+      return null;
+    }
+
+    return new ConfigObj(uri, obj);
+  }
+
+  /**
+   * Import an ini file
+   * @param filePath
+   * @returns `object` if file read is successful, or `null` if file open has failed
+   *
+   */
+  private static importIni(filePath: string): object|null {
+    let configRaw: string;
+    try {
+      configRaw = fs.readFileSync(filePath, 'utf-8');
+    } catch (e) {
+      console.error(e);
+      return null;
+    }
+
+    // TODO check if toString() is required
+    return ini.parse(configRaw.toString());
+  }
+
+  /**
+   * Parse base models written in the ini object and return the absolute path.
+   *
+   * ABOUT MULTIPLE BASE MODELS
+   * onecc doesn't support multiple base models.
+   * However, OneExplorer will show the config node below multiple base models
+   * to prevent a case that users cannot find their faulty config files on ONE explorer.
```

Comment:
As far as I understand, you mean that we should show such falty one cfg file in the view even though that has error with base model, because that will give user chance to fix the cfg file. I wonder I understand your intention correctly.